### PR TITLE
The DJ

### PR DESCRIPTION
**Pull request by carlopault**
image

## Changelog
:cl:
add: A new (non-half assed) job has been added, the DJ.
add: The Vacant Offine above Cargo has been replaced with the DJ's office.
add: 3 new machines, new clothing, new items, and additions to existing items have been added for the DJ. 
add: Bluespace radio station can play music to every headset and station-bounced radio in the game, as long as they're on a mob. 
add: Disk burner can be used to burn pre-coded or custom music onto record disks, which can be played at the bluespace radio station. This gives the DJ the power to play custom midis to people, just like an admin.
add: Unce machine plays unces over and over. 
add: Autolathes can now print record disks
soundadd: added a bunch of new songs, expect waiting a few extra seconds to download them all
imageadd: Daft punk helmet sprites were created by INFRARED_BARON. He allowed me to use his sprites for this PR.
imageadd: All other new sprites were done by Mr. Epic Sax. Make sure to thank him.
imagedel: deleted some icons and images
spellcheck: fixed typos somewhere and made code better elsewhere
code: landmines work a bit differently now
/:cl:

## About The Pull Request
Created with the help of @Neo1w and people in coderbus. Don't merge if the changelog is still broken, I want players to see this change. Yoyobatty, tell me if there is anything wrong with my changelog that would cause it to be broke and not appear in-game.

## Why It's Good For The Game
New content good.

### CODE REVIEW

**Comment by byandaccount** [comment]
+2 been waiting ages for this to come in, good work man

**Comment by MrEpicSax** [comment]
+1 Don't thank me they're ass. 

Be sure to leave Carlos plenty of feedback in the suggestions channel when it is merged.

**Comment by xTrainx** [comment]
+1, looks neat, dj needs an uzi tho

**Review by SpaceLoveSs13 — DISMISSED** [review]

**Comment by SpaceLoveSs13** [comment]
Loved it. Get a maintainer to approve it and get it merged.

**Comment by carlopault** [comment]
image

@AnthonyJNgo you lost the bet

File: code/game/objects/items/devices/PDA/PDA_types.dm [hunk 1]
```diff
@@ -170,6 +170,23 @@
 	icon_state = "pda-roboticist"
 	default_cartridge = /obj/item/cartridge/roboticist
 
+/obj/item/pda/discjockey
+	name = "\improper DJ PDA"
+	icon_state = "pda-dj"
+	default_cartridge = /obj/item/cartridge/discjockey
+
+/obj/item/pda/discjockey/Initialize()
+	..()
```
Comment (yoyobatty) [inline code/game/objects/items/devices/PDA/PDA_types.dm]: should be `. = ..()`

Comment (carlopault) [reply code/game/objects/items/devices/PDA/PDA_types.dm]: why

File: code/game/objects/items/storage/fancy.dm [hunk 2]
```diff
@@ -330,6 +330,20 @@
...
+
+/obj/item/storage/fancy/cigarettes/random/Initialize()
+	. = ..()
+	var/static/banned_cigarettes = list(/obj/item/storage/fancy/cigarettes/cigars,
+										/obj/item/storage/fancy/cigarettes/cigars/cohiba,
+										/obj/item/storage/fancy/cigarettes/cigars/havana
+										) //No cigars for you, DJ.
+	var/real_type = pick(subtypesof(/obj/item/storage/fancy/cigarettes) - banned_cigarettes)
+	new real_type(loc)
+	qdel(src)
```
Comment (yoyobatty) [inline code/game/objects/items/storage/fancy.dm]: should be `return INITIALIZE_HINT_QDEL`

File: code/game/objects/items/storage/fancy.dm [hunk 3]
```diff
@@ -330,6 +330,20 @@
 	icon_state = "cohibacase"
 	spawn_type = /obj/item/clothing/mask/cigarette/cigar/havana
 
+/obj/item/storage/fancy/cigarettes/random
+	name = "random cigarette pack"
+	desc = "If you can read this, please contact your local code monkey."
+
+/obj/item/storage/fancy/cigarettes/random/Initialize()
+	. = ..()
```
Comment (yoyobatty) [inline code/game/objects/items/storage/fancy.dm]: replace `. = ..()` with `..()`

**Review by yoyobatty — CHANGES_REQUESTED** [review]
Please have a look through and make sure everything works because I know you very well Carlos.

**Comment by carlopault** [comment]
yes maam

**Comment by carlopault** [comment]
yoyobatty can you merge this

**Review by yoyobatty — APPROVED** [review]

**Pull request merged**